### PR TITLE
android: add build & test to travis.

### PR DESCRIPTION
**Pull request by tswast**
I change the .gitignore and add gradle-wrapper.jar, because it is needed for the [./gradlew command](http://developer.android.com/tools/building/building-cmdline.html) to work. [Gradle says](https://docs.gradle.org/current/userguide/gradle_wrapper.html) this file *should* be in version control.

Adds Android support to our multi-language Travis for https://github.com/GoogleCloudPlatform/cloud-vision/issues/25.

### CODE REVIEW

**Comment by tswast** [comment]
PTAL @jerjou @amygdala

File: travis.sh [hunk 1]
```diff
@@ -67,6 +67,59 @@ build_java_oracle8() {
   build_java
 }
 
+build_android() {
+  # Set up Gradle, then update to latest version.
+  # http://www.tothenew.com/blog/gradle-installation-in-ubuntu/
+  sudo apt-get -y install python-software-properties # for apt-add-repository
+  yes | sudo add-apt-repository ppa:cwchien/gradle
+  sudo apt-get update
```
Comment (jerjou) [inline travis.sh]: Probably do

    sudo apt-get -qq update

for cleanliness of logs

File: travis.sh [hunk 2]
```diff
@@ -67,6 +67,59 @@ build_java_oracle8() {
   build_java
 }
 
+build_android() {
+  # Set up Gradle, then update to latest version.
+  # http://www.tothenew.com/blog/gradle-installation-in-ubuntu/
+  sudo apt-get -y install python-software-properties # for apt-add-repository
+  yes | sudo add-apt-repository ppa:cwchien/gradle
```
Comment (jerjou) [inline travis.sh]: I believe you can also pass `-y` to add-apt-repository

**Comment by amygdala** [comment]
That LGTM.  thank you!  @jerjou , can you still take a quick look as well?

File: travis.sh [hunk 3]
```diff
@@ -67,6 +67,59 @@ build_java_oracle8() {
   build_java
 }
 
+build_android() {
```
Comment (jerjou) [inline travis.sh]: I wonder if it's worth using [apt-fast](https://github.com/ilikenwf/apt-fast) - it's supposed to be 26x faster than apt-get, and looks like it's just a shell script that performs the downloads in parallel..

File: travis.sh [hunk 4]
```diff
@@ -67,6 +67,59 @@ build_java_oracle8() {
...
+  sudo apt-get -y install zlib1g:i386
+  wget http://dl.google.com/android/android-sdk_r24.4.1-linux.tgz
+  tar xzf android-sdk_r24.4.1-linux.tgz
+  export ANDROID_HOME=$PWD/android-sdk-linux
+  export PATH="${PATH}:${ANDROID_HOME}/tools:${ANDROID_HOME}/platform-tools"
+  # Accept the license.
+  # http://stackoverflow.com/a/17863931/101923
+  sudo apt-get -y install expect
+  expect -c '
+  set timeout -1   ;
```
Comment (jerjou) [inline travis.sh]: I assume `-1` means no timeout? Probably better to set it to something unreasonably high, just so we get _some_ sort of error if something goes horribly wrong..

File: travis.sh [hunk 5]
```diff
@@ -67,6 +67,59 @@ build_java_oracle8() {
...
+    "Do you accept the license" { exp_send "y\r" ; exp_continue }
+    eof
+  }
+  '
+  # Build and test the app!
+  cd android/CloudVision
+  ./gradlew --console=plain assembleDebug
+  ./gradlew --console=plain lintDebug
+  ./gradlew --console=plain testDebugUnitTest
+  cd ../..
```
Comment (jerjou) [inline travis.sh]: FYI (at least in bash) you can use parentheses to run a bunch of commands in another directory, and it'll go back to the original one afterwards:

    (
    cd android/CloudVision
    ./gradlew ....
    ....
    )

That way you don't have to worry about / remember to return to the original dir.

File: travis.sh [hunk 6]
```diff
@@ -67,6 +67,59 @@ build_java_oracle8() {
...
+  expect {
+    "Do you accept the license" { exp_send "y\r" ; exp_continue }
+    eof
+  }
+  '
+  # Build and test the app!
+  cd android/CloudVision
+  ./gradlew --console=plain assembleDebug
+  ./gradlew --console=plain lintDebug
+  ./gradlew --console=plain testDebugUnitTest
```
Comment (jerjou) [inline travis.sh]: I would chain these gradle commands with `&&` - that way if any of them fail (eg return code != 0), that'll be surfaced to the caller of this function

File: travis.sh (hunk 3 above)
Comment (tswast) [inline travis.sh]: I'd prefer not to introduce any unnecessary dependencies. Android builds are taking 3 minutes right now, which isn't too bad. It's cocoapods that I wish I could speed up for the iOS builds.

Comment (jerjou) [inline travis.sh]: Makes sense.

[this bug](https://github.com/travis-ci/travis-ci/issues/3534) seems to think you can use a local cache to further speed things up just by adding some config tweaks. Might be worth it?

File: travis.sh [hunk 7]
```diff
@@ -125,12 +180,12 @@ build_swift_ios() {
...
-    "platform=iOS Simulator,name=iPhone 6,OS=9.2" # 64bit
-    "platform=iOS Simulator,name=iPad 2,OS=8.1" # 32bit
-    "platform=iOS Simulator,name=iPad Air,OS=9.2" # 64bit
-  )
+  # IOS_DESTINATIONS=(
+  #   "platform=iOS Simulator,name=iPhone 4s,OS=8.1" # 32bit
+  #   "platform=iOS Simulator,name=iPhone 6,OS=9.2" # 64bit
+  #   "platform=iOS Simulator,name=iPad 2,OS=8.1" # 32bit
+  #   "platform=iOS Simulator,name=iPad Air,OS=9.2" # 64bit
+  # )
```
Comment (jerjou) [inline travis.sh]: Did you mean to comment these out in this PR?

**Comment by jerjou** [comment]
LGTM

File: travis.sh (hunk 7 above)
Comment (tswast) [inline travis.sh]: I did. They aren't doing anything right now. I commented them out after your exit code comment.

Comment (jerjou) [inline travis.sh]: LGTM

**Pull request merged**